api/data.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/dp-legacy-cache-api/models"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+func (api *API) GetDataSets(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		log.Info(ctx, "Calling get datasets handler")
+
+		results, err := api.dataStore.GetDataSets(ctx)
+		if err != nil {
+			log.Error(ctx, "Error retrieving datasets from db:", err)
```

Comment:
logging events should be lower case and avoid punctuation as per the [logging standards](https://github.com/ONSdigital/dp-standards/blob/main/LOGGING_STANDARDS.md#effective-event-types)
```suggestion
			log.Error(ctx, "error retrieving datasets from db", err)
```
This also applies to all other `log.Error` , `log.Info` etc. lines in this service.